app/controllers/posts_controller.rb

```diff
@@ -5,8 +5,18 @@ class PostsController < ApplicationController
 
 
 def index
-	posts = Post.all.chronological
-	@post_details = Post.get_post_details(posts)
+	if current_user.role? :admin
```

Comment:
Same thing commented on the other PR-- move all this to the model layer. Controller code should be short.